spec/models/manageiq/providers/azure/cloud_manager/provision_spec.rb

```diff
@@ -0,0 +1,93 @@
+require "spec_helper"
+
+describe ManageIQ::Providers::Azure::CloudManager::Provision do
+  let(:provider) { FactoryGirl.create(:ems_azure_with_authentication) }
+  let(:template) { FactoryGirl.create(:template_azure, :ext_management_system => provider) }
+  let(:flavor)   { FactoryGirl.create(:flavor_azure) }
+
+  context "#create vm" do
+    subscription_id = "01234567890"
+    resource_group  = "test_resource_group"
+    type            = "microsoft.resources"
+    name            = "vm_1"
+    nic_id          = "nic_id_1"
+
+    before(:each) do
+      subject.source = template
+      allow(subject).to receive(:gather_storage_account_properties).and_return(%w("target_uri", "source_uri", "windows"))
+      allow(subject).to receive(:create_nic).and_return(nic_id)
+    end
+
+    context "#find_destination_in_vmdb" do
+      it "VM in same sub-class" do
+        vm = FactoryGirl.create(:vm_azure, :ext_management_system => provider)
+        expect(subject.find_destination_in_vmdb([subscription_id, resource_group, type, name])).to eq(vm)
+      end
+
+      it "VM in same sub-class with invalid parameters" do
+        expect(subject.find_destination_in_vmdb(["invalid_subscription_id", resource_group, type, name])).to be_nil
+      end
+
+      it "VM in different sub-class" do
+        vm = FactoryGirl.create(:vm_openstack, :ext_management_system => provider)
+        vm.ems_ref = "openstack_vm"
```

Comment:
You can set the `ems_ref` as part of the `FactoryGirl.create` call so you do not need this line.